### PR TITLE
[GEOS-9942] Remove dependency on xpp3, use standard StAX API

### PR DESCRIPTION
**Pull request by groldan**
Remove dependency on xpp3 and use the standard StAX API instead.

[xpp3](https://mvnrepository.com/artifact/xpp3/xpp3) is an XML pull parser. Last released version is 14 years old. The
JRE comes with a standard StAX pull parser since Java 1.6.

Relates to https://github.com/geotools/geotools/pull/3382

---

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

**Review by mprins — APPROVED** [review]

File: src/ows/src/main/java/org/geoserver/ows/Dispatcher.java [hunk 1]
```diff
@@ -1525,51 +1515,34 @@ Object parseRequestKVP(Class type, Request request) throws Exception {
...
-            if ("service".equals(parser.getAttributeName(i))) {
-                service = parser.getAttributeValue(i);
-            }
-        }
-
-        parser.setInput(null);
-
-        // reset input stream
-        input.reset();
+        String namespace = request.getNamespace();
```
Comment (aaime) [inline src/ows/src/main/java/org/geoserver/ows/Dispatcher.java:1533]: This gave me pause at the beginning, but indeed, service (and thus readOpPost) are called before this section of code is reached, so all the the values are already set in the request, as parsed from the XML.

File: src/ows/src/main/java/org/geoserver/ows/Dispatcher.java [hunk 2]
```diff
@@ -1590,51 +1563,83 @@ Object parseRequestXML(Object requestBean, BufferedReader input, Request request
...
-            }
-
-            if ("outputFormat".equals(attName)) {
-                map.put("outputFormat", parser.getAttributeValue(i));
+        XMLInputFactory factory = XMLInputFactory.newFactory();
+        // disable DTDs
+        factory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+        // disable external entities
+        factory.setProperty(XMLInputFactory.IS_SUPPORTING_EXTERNAL_ENTITIES, false);
+        XMLStreamReader parser = factory.createXMLStreamReader(new CharArrayReader(buff));
```
Comment (aaime) [inline src/ows/src/main/java/org/geoserver/ows/Dispatcher.java:1610]: Shouldn't this reader be closed?

File: src/ows/src/main/java/org/geoserver/ows/Request.java [hunk 3]
```diff
@@ -48,6 +48,8 @@
     /** OWS request (ie operation) combined with service and version */
     protected String request;
 
+    protected String postRequestEementName;
```
Comment (aaime) [inline src/ows/src/main/java/org/geoserver/ows/Request.java:51]: Eement?

It's used in a lots of places, should be renamed with a refactor and make sure it's fixed everywhere.

File: src/ows/src/main/java/org/geoserver/ows/Request.java [hunk 4]
```diff
@@ -302,6 +304,14 @@ public void setRequest(String request) {
         this.request = request;
     }
 
+    String getPostRequestEementName() {
```
Comment (aaime) [inline src/ows/src/main/java/org/geoserver/ows/Request.java:307]: Bit of javadoc?

**Review by aaime — CHANGES_REQUESTED** [review]
Overall looks pretty good. Found a typo that really needs fixing, along with a few minor things.

File: src/ows/src/main/java/org/geoserver/ows/Dispatcher.java (hunk 2 above)
Comment (groldan) [reply src/ows/src/main/java/org/geoserver/ows/Dispatcher.java:1610]: no real need to, as it's running of a `CharArrayReader(char[])`, but still, won't hurt, so closing it

File: src/main/src/main/java/org/geoserver/catalog/SLDHandler.java [hunk 5]
```diff
@@ -343,28 +345,31 @@ public Version version(Object input) throws IOException {
...
+                if ("version".equals(parser.getAttributeLocalName(i))) {
                     version = parser.getAttributeValue(i);
                 }
             }
-
-            parser.setInput(null);
-        } catch (XmlPullParserException e) {
-            throw (IOException) new IOException("Error parsing content").initCause(e);
+            // release parser resources, does not close input stream
+            parser.close();
```
[the remark this answers is not in the document]
Comment (groldan) [reply src/main/src/main/java/org/geoserver/catalog/SLDHandler.java]: https://github.com/geoserver/geoserver/pull/4804/commits/3e71b504eb09a4dae97be299140e605285281134#diff-c94ec07f3c4d55a86727f5e790aaf2c78ed8423712af04717cab6b74f5d82d75R378-R383

File: src/ows/src/main/java/org/geoserver/ows/Dispatcher.java (hunk 2 above)
Comment (groldan) [reply src/ows/src/main/java/org/geoserver/ows/Dispatcher.java]: https://github.com/geoserver/geoserver/pull/4804/commits/86998b5ed3b342f44e2387a50b28077b37d3b16e#diff-09c0707df546d28c44971393a12fb1318901427a2f7957becdc5e69ecd315ee7R1595-R1619

File: src/ows/src/main/java/org/geoserver/ows/Request.java (hunk 3 above)
Comment (groldan) [reply src/ows/src/main/java/org/geoserver/ows/Request.java]: https://github.com/geoserver/geoserver/pull/4804/commits/86998b5ed3b342f44e2387a50b28077b37d3b16e#diff-be2ee09ab4b472e5452386650f99cd7c3381ff0c97c98396b39cb048212b7b01R308-R325

File: src/ows/src/main/java/org/geoserver/ows/Request.java (hunk 4 above)
Comment (groldan) [reply src/ows/src/main/java/org/geoserver/ows/Request.java]: https://github.com/geoserver/geoserver/pull/4804/commits/86998b5ed3b342f44e2387a50b28077b37d3b16e#diff-be2ee09ab4b472e5452386650f99cd7c3381ff0c97c98396b39cb048212b7b01R308-R325

**Comment by jodygarnett** [comment]
I am noticing  a small side effect of this change in https://osgeo-org.atlassian.net/browse/GEOS-10301 - another parser gets in the environment on accident and the stream is closed.

**Pull request merged**